core-overlay.html

```diff
@@ -111,666 +124,373 @@
+         * The backdrop element is attached to the document body and may be styled
+         * with the class `core-overlay-backdrop`. When opened the `core-opened`
+         * class is applied.
+         *
+         * @attribute backdrop
+         * @type boolean
+         * @default false
+         */
+        backdrop: {
+          type: Boolean,
+          value: false
+        },
 
-      /**
-       * A `core-overlay`'s size is guaranteed to be 
-       * constrained to the window size. To achieve this, the sizingElement
-       * is sized with a max-height/width. By default this element is the 
-       * target element, but it can be specifically set to a specific element
-       * inside the target if that is more appropriate. This is useful, for 
-       * example, when a region inside the overlay should scroll if needed.
-       *
-       * @attribute sizingTarget
-       * @type Object
-       * @default the target element
-       */
-      sizingTarget: null,
-    
-      /**
-       * Set opened to true to show an overlay and to false to hide it.
-       * A `core-overlay` may be made initially opened by setting its
-       * `opened` attribute.
-       * @attribute opened
-       * @type boolean
-       * @default false
-       */
-      opened: false,
+        /**
+         * If true, the ESC key does not cancel the overlay.
+         *
+         * @attribute noCancelOnEscKey
+         * @type boolean
+         * @default false
+         */
+        noCancelOnEscKey: {
+          type: Boolean,
+          value: false
+        },
 
-      /**
-       * If true, the overlay has a backdrop darkening the rest of the screen.
-       * The backdrop element is attached to the document body and may be styled
-       * with the class `core-overlay-backdrop`. When opened the `core-opened`
-       * class is applied.
-       *
-       * @attribute backdrop
-       * @type boolean
-       * @default false
-       */    
-      backdrop: false,
+        /**
+         * If true, clicking outside the overlay does not cancel the overlay.
+         *
+         * @attribute noCancelOnOutsideClick
+         * @type boolean
+         * @default false
+         */
+        noCancelOnOutsideClick: {
+          type: Boolean,
+          value: false
+        }
 
-      /**
-       * If true, the overlay is guaranteed to display above page content.
-       *
-       * @attribute layered
-       * @type boolean
-       * @default false
-      */
-      layered: false,
-    
-      /**
-       * By default an overlay will close automatically if the user
-       * taps outside it or presses the escape key. Disable this
-       * behavior by setting the `autoCloseDisabled` property to true.
-       * @attribute autoCloseDisabled
-       * @type boolean
-       * @default false
-       */
-      autoCloseDisabled: false,
+      },
+
+      get backdropElement() {
+        return this._backdrop;
+      },
+
+      registered: function() {
```

Comment:
Is `registered` a custom lifecycle callback of some sort? Where can I learn more about it?